haystack/retriever/dense.py

```diff
@@ -130,36 +116,56 @@ def embed_passages(self, texts: List[str], titles: Optional[List[str]] = None) -
         :param titles: passage title to also take into account during embedding
         :return: embeddings, one per input passage
         """
-        result = self._generate_batch_predictions(texts=texts, titles=titles, model=self.passage_encoder,
-                                                  tensorizer=self.tensorizer, batch_size=self.batch_size)
+        result = self._generate_batch_predictions(texts=texts, titles=titles,
+                                                  model=self.passage_encoder,
+                                                  tokenizer=self.passage_tokenizer,
+                                                  batch_size=self.batch_size)
         return result
 
+    def _normalize_question(self, question: str) -> str:
+        if question[-1] == '?':
```

Comment:
Is that how DPR was trained or why do we remove the "?" here? Maybe add a quick comment here to explain.

Reply:
DPR original code has this functionality in training, I copied the code from their `normalize_question` function